.github/workflows/deploy.yml

```diff
@@ -0,0 +1,93 @@
+name: Deploy API
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: 'Environment'
+        required: true
+        default: 'intg'
+      to_deploy:
+        description: 'Version to deploy'
+        required: true
+
+permissions:
+  id-token: write
+  contents: read
+  actions: write
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    environment: ${{ github.event.inputs.environment }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set environment specific account name
+        env:
+          ENVIRONMENT: ${{ github.event.inputs.environment }}
+        run: |
+          echo account_secret_name=${ENVIRONMENT^^}_ACCOUNT >> $GITHUB_ENV
+          echo environment_title=${ENVIRONMENT^} >> $GITHUB_ENV
+      - name: Configure AWS credentials for ECR
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: arn:aws:iam::${{ secrets.MANAGEMENT_ACCOUNT }}:role/GithubActionsTest
+          aws-region: eu-west-2
+          role-session-name: APIECRLogin
+      - name: Login to ECR
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ secrets.MANAGEMENT_ACCOUNT }}.dkr.ecr.eu-west-2.amazonaws.com
+      - name: Tag image
+        run: |
+          export MANIFEST=$(aws ecr batch-get-image --repository-name consignment-api --image-ids imageTag=${{ github.event.inputs.to_deploy }} --output json | jq --raw-output '.images[0].imageManifest')
+          aws ecr put-image --repository-name consignment-api --image-tag ${{ github.event.inputs.environment }} --image-manifest "$MANIFEST"
+      - name: Send tag created message
+        uses: slackapi/slack-github-action@v1.18.0
+        with:
+          payload: |
+            {
+              "message": "*Consignment API* :whale: The '${{ github.event.inputs.to_deploy }}' image has been tagged with '${{ github.event.inputs.environment }}' in ECR"
+            }
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_SUCCESS_WORKFLOW }}
+      - name: Configure AWS credentials for ECS
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: arn:aws:iam::${{ secrets[env.account_secret_name] }}:role/TDRGitHubECSUpdateRole${{ env.environment_title }}
+          aws-region: eu-west-2
+          role-session-name: APIECRLogin
+      - name: Deploy to ECS
+        run: aws ecs update-service --service consignmentapi_service_${{ github.event.inputs.environment }} --cluster consignmentapi_${{ github.event.inputs.environment }} --force-new-deployment
+      - name: Send service updated message
+        uses: slackapi/slack-github-action@v1.18.0
```

Comment:
Can this version `v1.18.0` be stored as a param. It's also used in the test yaml?

Reply:
Not really. Once I've got this working I'm going to have a proper look at creating some custom actions for these so we can reuse them and then it'll be in the same place then but for now I think it's ok to live with it.